plugins/bigquery/dbt/adapters/bigquery/impl.py

```diff
@@ -143,9 +143,27 @@ def truncate_relation(self, relation: BigQueryRelation) -> None:
     def rename_relation(
         self, from_relation: BigQueryRelation, to_relation: BigQueryRelation
     ) -> None:
-        raise dbt.exceptions.NotImplementedException(
-            '`rename_relation` is not implemented for this adapter!'
-        )
+        self.cache_renamed(from_relation, to_relation)
+
+        conn = self.connections.get_thread_connection()
+        client = conn.handle
+
+        from_table_ref = self.connections.table_ref(from_relation.database,
+                                                    from_relation.schema,
+                                                    from_relation.identifier,
+                                                    conn)
+        from_table = client.get_table(from_table_ref)
```

Comment:
Should this also ensure that `from_relation.type` and `to_relation.type` are the same type? And that `to_relation.type` is not `RelationType.View`?